src/spyglass/lfp/v1/lfp.py

```diff
@@ -260,6 +264,287 @@ def get_lfp_object(key: dict):
+    artifact_removed_valid_times: longblob
+    artifact_times: longblob # np array of artifact intervals
+    """
+
+
+def _get_artifact_times(
+    recording: None,
+    key: None,
+    zscore_thresh: Union[float, None] = None,
+    amplitude_thresh: Union[float, None] = None,
+    proportion_above_thresh: float = 1.0,
+    removal_window_ms: float = 1.0,
+):
+    """Detects times during which artifacts do and do not occur.
+    Artifacts are defined as periods where the absolute value of the change in LFP (ad units) exceeds one
+    or both specified amplitude or zscore thresholds on the proportion of channels specified,
+    with the period extended by the removal_window_ms/2 on each side. Z-score and amplitude
+    threshold values of None are ignored.
+
+    Parameters
+    ----------
+    recording : lfp eseries
+    zscore_thresh : float, optional
+        Stdev threshold for exclusion, should be >=0, defaults to None
+    amplitude_thresh : float, optional
+        Amplitude (ad units) threshold for exclusion, should be >=0, defaults to None
+    proportion_above_thresh : float, optional, should be>0 and <=1
+        Proportion of electrodes that need to have threshold crossings, defaults to 1
+    removal_window_ms : float, optional
+        Width of the window in milliseconds to mask out per artifact
+        (window/2 removed on each side of threshold crossing), defaults to 1 ms
+
+    Returns
+    -------
+    artifact_removed_valid_times : np.ndarray
+        Intervals of valid times where artifacts were not detected, unit: seconds
+    artifact_intervals : np.ndarray
+        Intervals in which artifacts are detected (including removal windows), unit: seconds
+    """
+
+    valid_timestamps = recording.timestamps
+
+    # if both thresholds are None, we skip artifact detection
+    if (amplitude_thresh is None) and (zscore_thresh is None):
+        recording_interval = np.asarray([valid_timestamps[0], valid_timestamps[-1]])
+        artifact_times_empty = np.asarray([])
+        print(
+            "Amplitude and zscore thresholds are both None, skipping artifact detection"
+        )
+        return recording_interval, artifact_times_empty
+
+    # verify threshold parameters
+    (
+        amplitude_thresh,
+        zscore_thresh,
+        proportion_above_thresh,
+    ) = _check_artifact_thresholds(
+        amplitude_thresh, zscore_thresh, proportion_above_thresh
+    )
+
+    # want to detect frames without parallel processing
+    # compute the number of electrodes that have to be above threshold
+    nelect_above = np.ceil(proportion_above_thresh * recording.data.shape[1])
+
+    # find the artifact occurrences using one or both thresholds, across channels
+    # replace with LFP artifact code
+    # is this supposed to be indices??
+    if (amplitude_thresh is not None) and (zscore_thresh is None):
+        artifact_boolean = np.sum(
+            (np.abs(np.diff(recording.data, axis=0)) > amplitude_thresh), axis=1
```

Comment:
What's the logic behind looking at the absolute value of the diff rather than just the absolute value?